Bindings/Packages/com.virtualmaker.bindings/Runtime/Bindings.InputActions.cs

```diff
@@ -0,0 +1,109 @@
+using System;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+namespace VirtualMaker.Bindings
+{
+    public partial class Bindings
+    {
+        public void Bind(InputActionReference inputActionReference, Action performed, Action started = null, Action cancelled = null)
+            => Bind(inputActionReference, _ => performed?.Invoke(), _ => started?.Invoke(), _ => cancelled?.Invoke());
+
+        public void Bind<T>(InputActionReference inputActionReference, Property<T> property) where T : struct
+            => Bind<T>(inputActionReference, value => property.Value = value);
+
+        public void Bind<T>(InputActionReference inputActionReference, Property<T> performed, Property<T> started, Property<T> cancelled = null)
+            where T : struct
+        {
+            Bind<T>(inputActionReference, value => performed.Value = value);
+
+            if (started != null)
+            {
+                Bind<T>(inputActionReference, value => started.Value = value);
+            }
+
+            if (cancelled != null)
+            {
+                Bind<T>(inputActionReference, value => cancelled.Value = value);
+            }
+        }
+
+        public void Bind<T>(InputActionReference inputActionReference, Action<T> performed, Action<T> started = null, Action<T> cancelled = null)
+            where T : struct
+        {
+            var inputAction = inputActionReference.GetAction();
+            inputAction.Enable();
+            inputAction.performed += OnActionPerformed;
+
+            AddUnsubscriber(() =>
+            {
+                inputAction.Disable();
+                inputAction.performed -= OnActionPerformed;
+            });
+
+            if (started != null)
+            {
+                inputAction.started += OnActionStarted;
+                AddUnsubscriber(() => inputAction.started -= OnActionStarted);
+            }
+
+            if (cancelled != null)
+            {
+                inputAction.canceled += OnActionCanceled;
+                AddUnsubscriber(() => inputAction.canceled -= OnActionCanceled);
+            }
+
+            return;
+            void OnActionPerformed(InputAction.CallbackContext ctx) => performed(ctx.ReadValue<T>());
+            void OnActionStarted(InputAction.CallbackContext ctx) => started(ctx.ReadValue<T>());
+            void OnActionCanceled(InputAction.CallbackContext ctx) => cancelled(ctx.ReadValue<T>());
+        }
+
+        public void Bind(InputActionReference inputActionReference, Action<InputAction.CallbackContext> performed, Action<InputAction.CallbackContext> started = null, Action<InputAction.CallbackContext> cancelled = null)
+        {
+            var inputAction = inputActionReference.GetAction();
+            inputAction.Enable();
+            inputAction.performed += OnActionPerformed;
+
+            AddUnsubscriber(() =>
+            {
+                inputAction.Disable();
+                inputAction.performed -= OnActionPerformed;
```

Comment:
And does it still work with however we implement `GetAction` for InputSystem versions before `1.8`?

Reply:
What would have changed pre 1.8? I don't understand the concern there.

> When one of them is reset, it calls inputAction.Disable();. If the user performs the input action again, would the other _binding still be able to have the performed/started/cancelled actions they passed in invoked?

From what I've experienced, when you call GetAction it seems to give you a copy of the InputAction, and that instance can be enabled/disabled. Would be worth a test to ensure that it would not disable a difference input action reference, that gets the input action from the Input System.